test/e2e/config/azure-dev.yaml

```diff
@@ -148,6 +148,7 @@ providers:
 
 variables:
   AKS_KUBERNETES_VERSION: "latest"
+  AKS_KUBERNETES_VERSION_UPGRADE_FROM: "${AKS_KUBERNETES_VERSION_UPGRADE_FROM:-stable-1.23}"
```

Comment:
I think this is fine for now, but FYI we'll have to monitor when the mutable "latest" alias gets rev'd by AKS, as that will require us to rev our "from" version (I'm pretty sure AKS will not let you upgrade > 1 version ahead)

Reply:
I do have some concerns with this as it can allow our test to break due to AKS rolling out a new version (as versions become unsupported). This is especially an issue for maintaining tests on older cherry-pick branches where our tests will naturally break over time.

/hold

Reply:
something more resilient (just off the top of my head) would be to maybe determine version to upgrade from as latest -1 (whether that's -1 minor or patch)

Reply:
That should work (subtract 1 from latest minor, that is). Using patch versions for testing upgrade can be tricky, as sometimes AKS will accelerate the deprecation of a patch version and leave a particular minor release (e.g., 1.25) with only one patch version available. (And there's also the case of a ".0" release, which will be the only patch available on that minor until the ".1" is released.)

Reply:
For sure, I just meant take the latest available version at index len() -1 in the array of supported versions by AKS.